Nodejs/Product/Npm/ReaderPackageJsonSource.cs

```diff
@@ -17,12 +17,21 @@
 using System;
 using System.IO;
 using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
 
 namespace Microsoft.NodejsTools.Npm {
     public class ReaderPackageJsonSource : IPackageJsonSource {
         public ReaderPackageJsonSource(TextReader reader) {
             try {
-                Package = JsonConvert.DeserializeObject(reader.ReadToEnd());
+                var text = reader.ReadToEnd();
+                try {
+                    // JsonConvert and JObject.Parse exhibit slightly different behavior,
+                    // so fall back to JObject.Parse if JsonConvert does not properly deserialize
+                    // the object.
+                    Package = JsonConvert.DeserializeObject(text);
+                } catch (ArgumentException) {
+                    Package = JObject.Parse(text);
+                }

```

Comment:
Is this definitely safe to sometimes fallback to a different method? I see DeserializeObject and JObject.Parse both return different types (Object and JObject respectively).

Also, may the new JObject.Parse now throw different exceptions that those you are handling below? (i.e. do you need to add more handlers for any JObject.Parse failures).